### DISCUSSION TITLE
New Ideas ?

### DISCUSSION THREAD

**Discussion by Mimieam** [Mimieam/TabSorter2#41] [Ideas]
This issue is an open thread for anyone wanting to bring up/ discuss a new idea :)

**Comment by Mimieam** [comment]
**Idea:** Feedback on clicks
**how:** On a button click, display some indication  ( ✅❌) in the lower coner of the button on success or failure of a command.

**Comment by Mimieam** [comment]
**Idea**: Horizontal Split 
**How**: split a window top and bottom

**Comment by volo1st** [comment]
**Idea**: Optional sorting on pinned tabs
**How**: Add an option in the options panel so that user can choose if they want the 'pinned tabs' to be sorted or not when they click on "sort all"

**Comment by kdvlr** [comment]
Idea: Sort Google Docs
How: Currently it sorts  by docs.google.com . For just that prefix, maybe  allow  docs.google.com/spreadsheets, docs.google.com/documents, docs.google.com/presentations to be treated as separate domain

**Comment by dunesitus** [comment]
Keyboard shortcuts for every function, but I use sort most, so at least for this one.

**Comment by gitatmax** [comment]
The ability to deduplicate not just the current but any tabs, leaving only 1 of each duplicate tab open

**Comment by Mimieam** [comment]
received via email:

> To whom it may concern:
> Regarding tab sorter chrome extension: 
> 
> Can the code of the extension be modified to access the various tab management functions from right-click context menu in addition to the menu bar button? 
> 
> Thanks, Jon

**Comment by yoasif** [comment]
Firefox support? Someone here is interested: https://www.reddit.com/r/firefox/comments/fajy19/is_there_an_addon_that_allows_me_to_seperate_tabs/

**Comment by Mimieam** [comment]
Thanks @yoasif,  there is currently no plan for supporting Firefox, but if we get more requests for it we will consider it :)

**Comment by F50Ci37Y** [comment]
Option to change counter color 
Add title hint for buttons

**Comment by tsilvs** [comment]
> Firefox support? Someone here is interested: [r/firefox: Is there an addon that allows me to seperate tabs?](https://www.reddit.com/r/firefox/comments/fajy19/is_there_an_addon_that_allows_me_to_seperate_tabs/)

> Thanks @yoasif, there is currently no plan for supporting Firefox, but if we get more requests for it we will consider it :)

@Mimieam
**Idea**: +1 on Firefox support.
**How**: There is a cross-browser extension boilerplate [repo](https://github.com/EmailThis/extension-boilerplate). And an [article](https://extensionworkshop.com/documentation/develop/porting-a-google-chrome-extension/) on Mozilla's Extension Workshop. Hope it helps.

**Comment by tsilvs** [comment]
**Idea**: Automatic Tab Sorting
**How**: A checkbox / toggle near Sort buttons. [Sort Tabs Advanced](https://github.com/monomon/sort-tabs-advanced) for Firefox has this feature.

**Comment by Mimieam** [comment]
> **Idea**: Automatic Tab Sorting
> **How**: A checkbox / toggle near Sort buttons. [Sort Tabs Advanced](https://github.com/monomon/sort-tabs-advanced) for Firefox has this feature.

Hey @JeffRockatansky actually pure automatic sorting currently exists and can be toggled in the options - check the toggle: `Force active sort`.
I've been working on porting TS2 to Firefox using `web-ext` - so far the results are mixed, and the core of TS2 might need to be rewritten to be more cross-browser, TBD.

**Comment by Mimieam** [comment]
> Option to change counter color

Definitely possible 

> Add title hint for buttons

@fakhamatia Could I have some examples for hints? 
the buttons did seem clear to me but I might have overlook somethings :)

**Comment by tsilvs** [comment]
> Hey @JeffRockatansky actually pure automatic sorting currently exists and can be toggled in the options - check the toggle: `Force active sort`.

@Mimieam thanks! Didn't see it initially. Does it have options for how exactly to sort (e.g. by URL, by domain, by title, ascending, descending)? Can't check myself right now.

> I've been working on porting TS2 to Firefox using `web-ext` - so far the results are mixed, and the core of TS2 might need to be rewritten to be more cross-browser, TBD.

Happy to know that! Any ETA?

If you could actually share the source code here, we might be able to help with the port. Btw, can you look at #26?

**Comment by tsilvs** [comment]
**Idea**: Option to configute how GUI looks. I personally like less round and more compact buttons from older versions over more round, but dark theme in later versions looks better for me.
**How**: as two config menu items: one selector for shapes, one for colors

**Comment by Mimieam** [comment]
> @Mimieam thanks! Didn't see it initially. Does it have options for how exactly to sort (e.g. by URL, by domain, by title, ascending, descending)? Can't check myself right now.

It has by Url and Title - the domain/subdomain is done by defaults when selecting URL. and Currently, it only sorts ASC 

> > I've been working on porting TS2 to Firefox using `web-ext` - so far the results are mixed, and the core of TS2 might need to be rewritten to be more cross-browser, TBD.
>
> Happy to know that! Any ETA?

No strong ETA currently because of the push to Manifest version 3, which is kind of a mess right now. 

> If you could actually share the source code here, we might be able to help with the port. Btw, can you look at #26?

Just did :)

**Comment by liango2** [comment]
Option to sort within each group, and keep the first open order between groups?
open in turn: a.**g2**.com;b.g1.com;b.g2.com,a.**g1**.com
sort result:
a.**g2**.com,b .g2.com|a.**g1**.com,b.g1.com

**Comment by shawn-yee** [comment]
Add a whitelist to remove duplicate tabs
For example, I'm reading two parts of an article through two tabs (same url), or editing something, i don't want to deduplicate tabs

**Comment by FarisHijazi** [comment]
add a tooltip description when hovering over each button

sometimes I don't know what some buttons do, would be nice to have a tooltip instead of reading the docs

**Comment by Mimieam** [comment]
@FarisHijazi There actually were tooltips for every button but it quickly cluttered the space and the layout/position was getting in the way... I'm still looking into how to make the buttons more intuitive ( via renaming or maybe revisiting tooltips if I can find a design that's clean.

do you mind sharing which buttons you find most confusing?

**Comment by Mimieam** [comment]
Via email: "it would be nice to have a Copy button and shortcut, that would copy all URLs of the current window to the clipboard"

**Comment by xgrommx** [comment]
Please answer https://github.com/Mimieam/TabSorter2/issues/28#issuecomment-1221364668

**Comment by Mimieam** [comment]
> Please answer [#28 (comment)](https://github.com/Mimieam/TabSorter2/issues/28#issuecomment-1221364668)

Done :)

**Comment by FarisHijazi** [comment]
Please add a "group tabs" option, one that uses titles to cluster the repeated words together, cuz we can't just depend on the name of the website or path.

Sometimes i have some github pages open about project X, and i have a bunch of stackoverflow pages on project X

I want all the ones with X to be in one place (automatically)

**Comment by Mimieam** [comment]
received via email:
> Could you display how long a tab has been opened for and how many times it was visited?

**idea**: Page Stats
**How**:  use content script to inject an active page with relevant info such as:
 - open since: X hrs ago
 - visited: X times
 - total time on this page/domain? (not sure if this one is useful..)
